tests/octf/trace/TracingTest.cpp

```diff
@@ -48,29 +53,31 @@ class Event {
             , m_size(0) {
         m_size = sizeof(EventSeqId);
         m_buffer = new char[m_size];
-        setSeqId();
+        initializeBuffer(m_buffer, m_size);
     }
 
     Event(uint32_t size)
             : m_buffer(nullptr)
             , m_size(size) {
         if (m_size < sizeof(EventSeqId)) {
-            m_size = sizeof(EventSeqId);
+            throw octf::Exception("Invalid event size");
         }
 
         m_buffer = new char[m_size];
 
-        for (uint64_t i = 0; i < m_size; i++) {
-            m_buffer[i] = randomDistribution(randomGenerator);
-        }
-
-        setSeqId();
+        initializeBuffer(m_buffer, m_size);
     }
 
-    Event(void *buffer, uint32_t size)
+    Event(void *buffer, uint32_t size, bool genId)
             : m_buffer(nullptr)
             , m_size(size) {
+        if (m_size < sizeof(EventSeqId)) {
+            throw octf::Exception("Invalid event size");
```

Comment:
Can we assert, fail with GTEST macro?

Reply:
Assert and fail return a value, so can't use them in the constructor